### PR TITLE
Story Locking: only show if there's an actual lock user

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Fixes showing the lock dialog if there wasn't actually a lock user set.

## Relevant Technical Choices

Remove redundant `showDialog` state variable

## User-facing changes

Fixes this bug:

Screenshot 2021-10-05 at 10 22 57

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Open editor
2. Don't see lock message

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@swissspidy Why does this fix the problem? What was the original problem?

File: packages/wp-story-editor/src/components/postLock/postLock.js [hunk 1]
```diff
@@ -124,8 +122,7 @@ function PostLock() {
 
   useEffect(() => {
     if (enablePostLocking && showLockedDialog && currentUserLoaded) {
-      if (lockUser?.id !== currentUser.id) {
```
Comment (swissspidy) [inline packages/wp-story-editor/src/components/postLock/postLock.js:127]: > Why does this fix the problem? What was the original problem?

@spacedmonkey This part was the underlying problem.

`lockUser` is always an object, with `lockUser.id` being `0` if there is no user.

So `lockUser?.id !== currentUser.id` is always `true`

Comment (dreamofabear) [reply packages/wp-story-editor/src/components/postLock/postLock.js:127]: > lockUser is always an object

This is a bit confusing. Is making `lockUser` nullable possible?

> lockUser.id being 0 if there is no user.

Could we change the fallback `id` to `NaN`?

File: packages/wp-story-editor/src/components/postLock/postLock.js [hunk 2]
```diff
@@ -140,7 +137,7 @@ function PostLock() {
   // Register an event on user navigating away from current tab to release / delete lock.
   useEffect(() => {
     function releasePostLock() {
-      if (enablePostLocking && showLockedDialog && !showDialog && nonce) {
+      if (enablePostLocking && showLockedDialog && user?.id && nonce) {
```
Comment (dreamofabear) [inline packages/wp-story-editor/src/components/postLock/postLock.js:140]: Similar situation? `user` is always non-null but with falsy properties?

**Review by dreamofabear — APPROVED** [review]
Looks ok for a quick fix but I'd suggest some refactoring for better readability long-term:

1. Use null to represent "no user" or "no locked user" if possible
2. Use `NaN` or a random string for "invalid" property semantics to avoid unintentional truthy comparisons

**Comment by swissspidy** [comment]
> Use null to represent "no user" or "no locked user" if possible

Yeah that makes sense.

We have similar situations in the editor for things like story poster, where we use an "empty" object instead of null. Best to tackle that coherently in a dedicated ticket 👍

File: packages/wp-story-editor/src/components/postLock/postLock.js (hunk 1 above)
Comment (spacedmonkey) [inline packages/wp-story-editor/src/components/postLock/postLock.js:127]: Would not have been cleaner to do this? 
```
if (lockUser && lockUser.id !== currentUser.id) {
```

**Pull request merged**